File: SieveOfEratosthenes.py
#Program to print all prime numbers <=n using Sieve of Eratosthenes

def sieveOfEratosthenes(n):
    prime_nums=[True for i in range(n+1)] #array with all True values
    x=2
    
    while(x*x<=n):
        if prime_nums[x]==True:
            for i in range(x*x,n+1,x):
                prime_nums[i]=False #updating multiples of x to false
        x+=1    
        
    for x in range(2,n+1):
        if prime_nums[x]:
            print(x)
            
        
sieveOfEratosthenes(200)

File: SieveOfEratosthenes.java
import java.util.Scanner;

public class SieveOfEratosthenes {
    public static void sieve(int n) {
        // Step 1: Create a boolean array and initialize all entries as true
        boolean[] isPrime = new boolean[n + 1];
        for (int i = 2; i <= n; i++) {
            isPrime[i] = true;
        }

        // Step 2: Start marking non-primes
        for (int p = 2; p * p <= n; p++) {
            // If isPrime[p] is not changed, then it is a prime
            if (isPrime[p]) {
                // Update all multiples of p as not prime
                for (int i = p * p; i <= n; i += p) {
                    isPrime[i] = false;
                }
            }
        }

        // Step 3: Print all prime numbers
        System.out.println("Prime numbers up to " + n + ":");
        for (int i = 2; i <= n; i++) {
            if (isPrime[i]) {
                System.out.print(i + " ");
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        int n = sc.nextInt();
        sieve(n);
        sc.close();
    }
}
